src/client/shader.cpp

```diff
@@ -360,6 +363,12 @@ ShaderSource::ShaderSource()
 	// Add a dummy ShaderInfo as the first index, named ""
 	m_shaderinfo_cache.emplace_back();
 
+	m_enabled = g_settings->getBool("enable_shaders");
+	if (!m_enabled) {
+		warningstream << "You are running " PROJECT_NAME_C " with shaders disabled, "
```

Comment:
While I'm a big fan of this, isn't this a bit heavy-handed?
Is this is preparation to remove the option to disable shaders in the future?

Reply:
Exactly. Shaders are a basic part of rendering, not something you disable for any reason.